server/src/controllers/callback.controller.ts

```diff
@@ -0,0 +1,91 @@
+import { Request, Response, Router } from "express";
+import { MongoClient } from "mongodb";
+import { DI } from "../server";
+import qs from 'qs';
+async function createUser(user: any): Promise<any> {
+  const uri = "mongodb+srv://testuser:randompassword123@cluster0.9b1cx.mongodb.net/spotify?retryWrites=true&w=majority";
+  const client = new MongoClient(uri); 
+  console.log("Connecting to MongoDB");
+  try{
+    await client.connect();
+    console.log("Connected to MongoDB");
+    const result = await client.db("spotify").collection("users").insertOne(user);
+  } catch (e) {
+    console.log(e);
+  } finally {
+    await client.close();
+  }
+}
+var axios = require('axios');
+var cookieParser = require('cookie-parser');
+const router = Router();
+router.use(cookieParser());
```

Comment:
I would probably do this in server.ts, so we can keep track of all of our middleware more easily